tests/test_build_utils.py

```diff
@@ -38,3 +42,112 @@ def test_input_output_name_backwards_compatibility(dtype):
         "output_values",
         "output_null",
     ]
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        ndx.int8,
+        ndx.int16,
+        ndx.int32,
+        ndx.int64,
+        ndx.float32,
+        ndx.float64,
+        ndx.utf8,
+        ndx.bool,
+        ndx.uint8,
+        ndx.uint16,
+        ndx.uint32,
+        ndx.uint64,
+    ],
+)
+def test_no_namemangling_for_standard_types(dtype):
+    a = ndx.array(shape=("N",), dtype=dtype)
+    model_proto = ndx.build({"input": a}, {"output": a})
+    assert [node.name for node in model_proto.graph.input] == ["input"]
+    assert [node.name for node in model_proto.graph.output] == ["output"]
+    a = ndx.array(shape=("N",), dtype=dtypes.into_nullable(dtype))
+    model_proto = ndx.build({"input": a}, {"output": a})
+    assert {node.name for node in model_proto.graph.input} == {
+        "input_values",
+        "input_null",
+    }
+    assert {node.name for node in model_proto.graph.output} == {
+        "output_values",
+        "output_null",
+    }
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        ndx.bool,
+        ndx.float32,
+        ndx.float64,
+        ndx.int16,
+        ndx.int32,
+        ndx.int64,
+        ndx.int8,
+        ndx.nbool,
+        ndx.nfloat32,
+        ndx.nfloat64,
+        ndx.nint16,
+        ndx.nint32,
+        ndx.nint64,
+        ndx.nint8,
+        ndx.nuint16,
+        ndx.nuint32,
+        ndx.nuint64,
+        ndx.nuint8,
+        ndx.nutf8,
+        ndx.uint16,
+        ndx.uint32,
+        ndx.uint64,
+        ndx.uint8,
+        ndx.utf8,
+    ],
+)
+def test_schema_against_snapshots(dtype):
+    # We must not break backwards compatibility. We test every type we
+    # support that it keeps producing the same schema.
+
+    a = ndx.array(shape=("N",), dtype=dtype)
+    b = a[0]  # make the build non-trivial
+
+    mp = ndx.build({"a": a}, {"b": b})
+
+    # These files should not be update automatically!
+    # File names follow NumPy's __str__ semantics for primitive data
+    # types and generally uses lower case. ndonnx does not have the
+    # same __str__ semantics today.
+    if dtype == ndx.bool:
+        dtype = "bool"
+    elif dtype == ndx.nbool:
+        dtype = "nbool"
+    else:
+        dtype = str(dtype).lower()
+    fname = Path(__file__).parent / f"schemas/{dtype}.json"
+
+    # Only set to `True` temporarily if the schemas need to be
+    # re-generated. A schema, once stabilized, must never
+    # change. However, this flag may come in handy if we add new data
+    # types for instance.
+    update = False
+    if update:
```

Comment:
I'd remove the update flag from the test suite and have this in an independent script. You can make a pixi task for generating schemas from the dtype submodule if you'd like some automation around this.

Reply:
I like having the code here since it directly hooks into the parametrization of the test. We would need to maintain a separate list of data types for the script along with the path etc. Also, running a pixi task is probably easier than we want the generation to be. I'd prefer to keep things like this.

Reply:
A pytest argument will do; just no source code manipulation to generate schemas please!